util/src/main/scala/io/catbird/util/asyncstream.scala

```diff
@@ -1,24 +1,27 @@
 package io.catbird
 package util
 
-import cats.{ Eq, Monad, Monoid, Semigroup, StackSafeMonad }
+import cats.{ Alternative, Eq, Monad, Monoid, Semigroup, StackSafeMonad }
 import com.twitter.concurrent._
 import com.twitter.util._
 
 trait AsyncStreamInstances extends AsyncStreamInstances1 {
-  implicit final val asyncStreamInstances: Monad[AsyncStream] =
-    new StackSafeMonad[AsyncStream] {
+  implicit final val asyncStreamInstances: Monad[AsyncStream] with Alternative[AsyncStream] =
+    new StackSafeMonad[AsyncStream] with Alternative[AsyncStream] {
       final def pure[A](a: A): AsyncStream[A] = AsyncStream.of(a)
       final def flatMap[A, B](fa: AsyncStream[A])(f: A => AsyncStream[B]): AsyncStream[B] = fa.flatMap(f)
       override final def map[A, B](fa: AsyncStream[A])(f: A => B): AsyncStream[B] = fa.map(f)
+
+      def empty[A]: AsyncStream[A] = AsyncStream.empty[A]
+      def combineK[A](x: AsyncStream[A], y: AsyncStream[A]): AsyncStream[A] = x ++ y
```

Comment:
Is there a rule/convention governing whether these should explicitly specify `override`, or be `final`?

Reply:
I personally like the explicit override modifier but in this project it's not used (afaik). `final` might have performance implications but I'm not an expert on it. 
In general just stick with what the existing code uses if there's no explicit styleguide.